repos/system_upgrade/common/actors/scansourcefiles/tests/unit_test_scansourcefiles.py

```diff
@@ -1,5 +1,175 @@
+    ({'exit_code': 1, 'stdout': 'S.5......  c /etc/openldap/ldap.conf'}, True),
+    ({'exit_code': 1, 'stdout': '..?......  c /etc/libaudit.conf'}, False),
+    ({'exit_code': 1, 'stdout': '.....UG..  g /var/run/avahi-daemon'}, False),
+))
+def test_is_modified(monkeypatch, run_output, expected_output):
+    def mocked_run(cmd, *args, **kwargs):
+        assert cmd == ['rpm', '-Vf', '--nomtime', input_file]
+        return run_output
+
+    input_file = '/file'
+    monkeypatch.setattr(scansourcefiles, 'run', mocked_run)
+
+    assert scansourcefiles.is_modified(input_file) == expected_output
+
+
+@pytest.mark.parametrize('run_output,expected_output', (
+    ({'stdout': ['rpm']}, 'rpm'),
+    ({'stdout': ['rpm1', 'rpm2']}, 'rpm1'),
+    (CalledProcessError, ''),
+))
+def test_get_rpm_name(monkeypatch, run_output, expected_output):
+    def mocked_run(cmd, *args, **kwargs):
+        assert cmd == ['rpm', '-qf', '--queryformat', r'%{NAME}\n', input_file]
+        if raise_error:
+            raise CalledProcessError("mocked error", cmd, "result")
+        return run_output
+
+    raise_error = run_output and not isinstance(run_output, dict)
+    input_file = '/file'
+
+    monkeypatch.setattr(scansourcefiles, 'run', mocked_run)
+    monkeypatch.setattr(api, 'current_logger', testutils.logger_mocked())
+
+    assert scansourcefiles._get_rpm_name(input_file) == expected_output
+
+    if raise_error:
+        return
+
+    if len(run_output['stdout']) > 1:
+        assert len(api.current_logger.warnmsg) == 1
+        expected_warnmsg = 'The {} file is owned by multiple rpms: rpm1, rpm2.'.format(input_file)
+        assert api.current_logger.warnmsg[0] == expected_warnmsg
+    else:
+        assert not api.current_logger.warnmsg
+
+
+@pytest.mark.parametrize('input_file,rpm_name,exists,modified,expected_output', (
+    ('/file1', 'rpm', False, True,
+     FileInfo(path='/file1', exists=False, rpm_name='rpm', is_modified=True)),
+
+    ('/file2', '', True, False,
+     FileInfo(path='/file2', exists=True, rpm_name='', is_modified=False))
+))
+def test_scan_file(monkeypatch, input_file, rpm_name, modified, exists, expected_output):
+    monkeypatch.setattr(scansourcefiles, 'is_modified', lambda _: modified)
+    monkeypatch.setattr(scansourcefiles, '_get_rpm_name', lambda _: rpm_name)
+    monkeypatch.setattr(os.path, 'exists', lambda _: exists)
+
+    file_info = scansourcefiles.scan_file(input_file)
+
+    assert isinstance(file_info, FileInfo)
+    assert file_info == expected_output
+
+
+@pytest.mark.parametrize('input_files,rpm_name,exists,modified,expected_output', (
+    ([], '', False, False, []),
+
+    (['/file1'], '', False, False,
+     [FileInfo(path='/file1', exists=False, rpm_name='', is_modified=False)]),
+
+    (['/file1', '/file2'], '', False, False,
+     [FileInfo(path='/file1', exists=False, rpm_name='', is_modified=False),
+      FileInfo(path='/file2', exists=False, rpm_name='', is_modified=False)]),
```

Comment:
that's kind of duplicate input data and also prevents testing combination of files that exists or not, that are owned by rpm or not. The first line specifies params that must be found in all later defined files + names of files.  You can obtain the same result with params defined on the first line, or just with the last param. The second way is what I would do, as It could also introduce testing for various combinations. Another benefit is that input data are more readable.

Reply:
Done.